src/include/index/index_group.h

```diff
@@ -176,13 +147,13 @@ class base_index_group {
       throw std::runtime_error(
           "Group uri " + std::string(group_uri_) + " does not exist.");
     }
-    auto read_group = tiledb::Group(cached_ctx_, group_uri_, TILEDB_READ, cfg);
+    read_group_ = new tiledb::Group(cached_ctx_, group_uri_, TILEDB_READ);
```

Comment:
If we use this style of Group creation we need to figure out a way to `close` and `delete` the group

Reply:
Won't this happen automatically when the base group is destructed and we trigger the read_group_ destructor?

Reply:
Okay no I was correct initially, and you're correct, it will not get deleted. Got confused by the comment by Andrew, but it's a raw pointer, so there will be no automatic deletion when it goes out of scope.

Reply:
Talked offline - will go back to the map and update to a group after fixing the bug in `cached_ctx_`. See full context here: https://tiledb.slack.com/archives/C0537B4V7Q8/p1713447475923239?thread_ts=1713354838.545549&cid=C0537B4V7Q8

Reply:
In the Slack discussion I think we were talking about two different things.  Sorry.  I was talking about `read_group_` as a smart pointer, which was how the discussion thread started.

A raw pointer is non-owning (per C++ Core Guidelines R.3).  I don't know if it is an org requirement, but it should be.  Else for the `new` you have to do a `delete` etc.

Reply:
Ah I really got tricked there haha. I was like "well I definitely didn't think this, but I know Andrew knows C++ 😅".